File: file_formats_bonanza [INDIVIDUAL]/python/file_reader.py
################# TXT FILE READER ###################

with open('../text_file.txt') as f:
        lines = f.read()
        print(lines)

################# CSV FILE READER ###################

import csv

with open('../csv_file.csv') as csv_file:
    reader = csv.reader(csv_file)

    for row in reader:
        print(row)
    csv_file.close()

################# JSON FILE READER ###################

import json

with open('../json_file.json', 'r') as json_content:
    print(json.load(json_content))

################# XML FILE READER ###################

import xml.etree.ElementTree as ET

root = ET.parse('..\\xml_file.xml').getroot()
print(root[0].text)

################# YAML FILE READER ###################

import yaml 

with open('../yaml_file.yaml') as yaml_file:
    print(yaml.safe_load(yaml_file))

File: file_formats_bonanza [INDIVIDUAL]/node/file_reader.js
/////////////////////// TEXT FILE READER ///////////////////////////////

const fs = require('fs');

try {
    let data = fs.readFileSync('../text_file.txt', 'utf-8');
    console.log(data.toString());
} catch (e) {
    console.log('Error: ', e.stack);
}

/////////////////////// CSV FILE READER ///////////////////////////////

const csv = require('fast-csv');

fs.createReadStream('../csv_file.csv')
    .pipe(csv.parse({ headers: true }))
    .on('error', error => console.error(error))
    .on('data', row => console.log(row));

/////////////////////// JSON FILE READER ///////////////////////////////

fs.readFile('../json_file.json', 'utf-8', function(err, data) {
    if(err) throw(err);
    console.log(JSON.parse(data));
});

/////////////////////// XML FILE READER ///////////////////////////////

const parser = require('xml-js');

fs.readFile('../xml_file.xml', 'utf-8', function(err, data) {
    if(err) throw(err);
    console.log(parser.xml2json(data, {compact: true, spaces: 2}));
});

/////////////////////// YAML FILE READER ///////////////////////////////

const yaml = require('js-yaml');

fs.readFile('../yaml_file.yaml', 'utf-8', function(err, data) {
    if(err) throw(err);
    console.log(yaml.load(data));
});